tests/test-zuul-jobs.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash -xe
+# This tests the JJB syntax of our job definitions, and generates a
+# job-list.txt for use in the layout-checks.py test.
+
+jobs_dir="roles/zuul/files/jobs/"
+jobs_out=$1
+if [ -z $jobs_out ]; then
+    echo "ERROR: Must specify a file for jobs list output"
+    exit 1
+fi
+
+temp_dir=`mktemp -d`
+trap "rm -rf $temp_dir" EXIT
+
+virtualenv $temp_dir
+source $temp_dir/bin/activate
+pip install jenkins-job-builder
+jenkins-jobs --l debug test -o $temp_dir/jobs $jobs_dir
```

Comment:
You probably wanna run it through shellcheck while you're at it, as shown in the failed hoist-checks above.